src/inmanta/agent/forking_executor.py

```diff
@@ -443,7 +444,7 @@ async def call(self, context: ExecutorContext) -> FailedInmantaModules:
             except Exception as e:
                 logger.info("Failed to import source: %s", module_source.metadata.name, exc_info=True)
                 inmanta_module_name = module_source.get_inmanta_module_name()
-                failed[inmanta_module_name][module_source.metadata.name] = e
+                failed[inmanta_module_name][module_source.metadata.name] = ModuleImportException(e, module_source.metadata.name)
```

Comment:
I wonder if this `failed` dict could be changed to be a dict of `str` -> `str` -> `str` instead of `str` -> `str` -> `Exception`

Reply:
Q2: Would it be ok to "serialize" these exceptions to strings ?

Reply:
Eventually, you have to, so it is more a question of when than if.

I have no good idea of where the best place would be.

Reply:
I'll leave it be for now then.